certification/internal/policy/operator/scc_info.go

```diff
@@ -0,0 +1,124 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/bundle"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+)
+
+// securityContextConstraintsCheck evaluates the csv and logs a message if a non default security context constraint is
+// needed by the operator
+type securityContextConstraintsCheck struct {
+	customSCCSpecified  bool
+	builtInSCCSpecified bool
+}
+
+func getDefaultSCCs() []string {
+	return []string{"restricted", "privileged", "nonroot", "node-exporter", "hostnetwork", "hostmount-anyuid", "hostaccess", "anyuid"}
+}
+
+func NewSecurityContextConstraintsCheck() *securityContextConstraintsCheck {
+	return &securityContextConstraintsCheck{
+		false,
+		false,
+	}
+}
+
+func (p *securityContextConstraintsCheck) Validate(ctx context.Context, bundleRef certification.ImageReference) (bool, error) {
+	requestedSCCList, err := p.dataToValidate(ctx, bundleRef.ImageFSPath)
+	if err != nil {
+		return false, err
+	}
+
+	return p.validate(ctx, requestedSCCList)
+}
+
+func (p *securityContextConstraintsCheck) dataToValidate(ctx context.Context, imagePath string) ([]string, error) {
+	csvFilepath, err := bundle.GetCsvFilePathFromBundle(imagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	csvFileReader, err := os.Open(csvFilepath)
+	if err != nil {
+		return nil, err
+	}
+
+	requestedSccList, err := bundle.GetSecurityContextConstraints(ctx, csvFileReader)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract security context constraints from ClusterServiceVersion: %w", err)
+	}
+
+	return requestedSccList, nil
+}
+
+func (p *securityContextConstraintsCheck) validate(ctx context.Context, requestedSccList []string) (bool, error) {
+	defaultSccList := getDefaultSCCs()
+
+	if len(requestedSccList) == 0 {
+		log.Infof("No custom security context constraint was detected in the csv the default the restricted scc will be used")
+		return true, nil
+	}
+
+	if len(requestedSccList) != 1 {
+		return false, fmt.Errorf("only one scc should be requested at a time")
+	}
+
+	for _, defaultScc := range defaultSccList {
```

Comment:
You could avoid this loop if you just made the default SCCs a map instead of a list. map[string]struct{} would do the trick. Then you could just check if the requested SCC is in the map as a key. i probably wouldn't even bother with the 'getDefaultsSccs' function, and just have the map unexported as a var.

Reply:
removed entirely since we only need to print out the scc and we dont care if its custom or default..